consensus/types/src/light_client_header.rs

```diff
@@ -1,26 +1,223 @@
+            Decode,
+            Encode,
+            TestRandom,
+            arbitrary::Arbitrary,
+        ),
+        serde(bound = "E: EthSpec", deny_unknown_fields),
+        arbitrary(bound = "E: EthSpec"),
+    )
 )]
-pub struct LightClientHeader {
+#[derive(Debug, Clone, Serialize, Deserialize, arbitrary::Arbitrary, PartialEq)]
+#[serde(untagged)]
+// #[ssz(enum_behaviour = "transparent")]
+#[serde(bound = "E: EthSpec", deny_unknown_fields)]
+#[arbitrary(bound = "E: EthSpec")]
+pub struct LightClientHeader<E: EthSpec> {
     pub beacon: BeaconBlockHeader,
+    #[superstruct(only(Capella, Deneb))]
+    pub execution_branch: FixedVector<Hash256, ExecutionPayloadProofLen>,
+    #[superstruct(
+        only(Capella),
+        partial_getter(rename = "execution_payload_header_capella")
+    )]
+    pub execution: ExecutionPayloadHeaderCapella<E>,
+    #[superstruct(only(Deneb), partial_getter(rename = "execution_payload_header_deneb"))]
+    pub execution: ExecutionPayloadHeaderDeneb<E>,
+
+    #[serde(skip)]
+    #[ssz(skip_serializing, skip_deserializing)]
+    pub phantom_data: PhantomData<E>,
+}
+
+impl<E: EthSpec> LightClientHeader<E> {
+    pub fn block_to_light_client_header(
+        block: &SignedBeaconBlock<E>,
+        fork_name: ForkName,
+    ) -> Result<Self, Error> {
+        let header = match fork_name {
+            ForkName::Base => return Err(Error::AltairForkNotActive),
+            ForkName::Altair | ForkName::Merge => LightClientHeader::Altair(
+                LightClientHeaderAltair::block_to_light_client_header(block)?,
+            ),
+            ForkName::Capella => LightClientHeader::Capella(
+                LightClientHeaderCapella::block_to_light_client_header(block)?,
+            ),
+            ForkName::Deneb => LightClientHeader::Deneb(
+                LightClientHeaderDeneb::block_to_light_client_header(block)?,
+            ),
+        };
+        Ok(header)
+    }
+
+    pub fn from_ssz_bytes(bytes: &[u8], fork_name: ForkName) -> Result<Self, ssz::DecodeError> {
+        let header = match fork_name {
+            ForkName::Altair | ForkName::Merge => {
+                let header = LightClientHeaderAltair::from_ssz_bytes(bytes)?;
+                LightClientHeader::Altair(header)
+            }
+            ForkName::Capella => {
+                let header = LightClientHeaderCapella::from_ssz_bytes(bytes)?;
+                LightClientHeader::Capella(header)
+            }
+            ForkName::Deneb => {
+                let header = LightClientHeaderDeneb::from_ssz_bytes(bytes)?;
+                LightClientHeader::Deneb(header)
+            }
+            ForkName::Base => {
+                return Err(ssz::DecodeError::BytesInvalid(format!(
+                    "LightClientHeader decoding for {fork_name} not implemented"
+                )))
+            }
+        };
+
+        Ok(header)
+    }
+}
+
+impl<E: EthSpec> Encode for LightClientHeader<E> {
+    fn is_ssz_fixed_len() -> bool {
```

Comment:
I think the `is_ssz_fixed_len` and `ssz_bytes_len` implementations may need to account for the fixed sized `Altair` variant?